v3/src/components/data-display/components/legend/color-legend.tsx

```diff
@@ -0,0 +1,61 @@
+import { reaction } from "mobx"
+import { observer } from "mobx-react-lite"
+import React, { useCallback, useEffect } from "react"
+import {axisGap} from "../../../axis/axis-types"
+import {getStringBounds} from "../../../axis/axis-utils"
+import {useDataConfigurationContext} from "../../hooks/use-data-configuration-context"
+
+import vars from "../../../vars.scss"
+
+interface IColorLegendProps {
+  layerIndex: number
+  setDesiredExtent: (layerIndex:number, extent: number) => void
+}
+
+export const ColorLegend =
+  observer(function ColorLegend({layerIndex, setDesiredExtent}: IColorLegendProps) {
+  const dataConfiguration = useDataConfigurationContext(),
+
+    getLabelHeight = useCallback(() => {
+      const labelFont = vars.labelFont,
+        legendAttrID = dataConfiguration?.attributeID('legend') ?? ''
+      return getStringBounds(dataConfiguration?.dataset?.attrFromID(legendAttrID)?.name ?? '', labelFont).height
+    }, [dataConfiguration]),
+
+    refreshScale = useCallback(() => {
+      const labelHeight = getLabelHeight()
+      const computeDesiredExtent = () => {
+        if (dataConfiguration?.placeCanHaveZeroExtent('legend')) {
+          return 0
+        }
+        return labelHeight + 2 * axisGap
+      }
+
+      setDesiredExtent(layerIndex, computeDesiredExtent())
+    }, [dataConfiguration, getLabelHeight, setDesiredExtent, layerIndex])
+
+  useEffect(function refresh() {
+    refreshScale()
+  }, [refreshScale])
+
+  useEffect(function respondToLayoutChange() {
+    const disposer = reaction(
```

Comment:
Could eliminate line 51 with:
```suggestion
    return reaction(
```